extensions/quartz/runtime/src/main/java/io/quarkus/quartz/runtime/QuartzScheduler.java

```diff
@@ -277,8 +277,8 @@ public QuartzScheduler(SchedulerContext context, QuartzSupport quartzSupport, Sc
                         org.quartz.Trigger trigger = triggerBuilder.build();
                         org.quartz.Trigger oldTrigger = scheduler.getTrigger(trigger.getKey());
                         if (oldTrigger != null) {
-                            scheduler.rescheduleJob(trigger.getKey(),
-                                    triggerBuilder.startAt(oldTrigger.getNextFireTime()).build());
+                            trigger = triggerBuilder.startAt(oldTrigger.getNextFireTime()).build();
```

Comment:
re-assign the trigger with the proper start time as it is used in https://github.com/quarkusio/quarkus/blob/c4aa8948da8bc78aa7560a7596b1898961aea0cc/extensions/quartz/runtime/src/main/java/io/quarkus/quartz/runtime/QuartzScheduler.java#L303